pkg/controller/sidecar/sync.go

```diff
@@ -134,5 +168,13 @@ func (c *Controller) sync(ctx context.Context) error {
 		}
 	}
 
+	hostIdValue, hasHostIdAnnotation := svc.Annotations[naming.HostIdAnnotation]
+	if !hasHostIdAnnotation || (c.hostId != nil && hostIdValue != *c.hostId) {
+		err := c.setHostIdLabel(ctx, svc)
```

Comment:
nit: I'd inline the set method but extract the `GetHostID` method that either uses cache or and API call. the condition above will be also simplified.